agent/pkg/containerbuilder/container_builder.go

```diff
@@ -11,194 +12,211 @@ import (
 	"github.com/docker/docker/api/types/mount"
 	"github.com/docker/docker/client"
 	"github.com/docker/go-connections/nat"
-
-	"github.com/dyrector-io/dyrectorio/agent/internal/dogger"
-	"github.com/dyrector-io/dyrectorio/agent/internal/util"
-	v1 "github.com/dyrector-io/dyrectorio/agent/pkg/api/v1"
-	"github.com/dyrector-io/dyrectorio/agent/pkg/dagent/config"
 )
 
 type ContainerBuilder interface {
 	WithClient(client *client.Client) ContainerBuilder
 	WithImage(imageName string) ContainerBuilder
 	WithEnv(env []string) ContainerBuilder
-	WithPortBindings(portList []v1.PortBinding) ContainerBuilder
-	WithPortRange(portRanges []v1.PortRange) ContainerBuilder
+	WithPortBindings(portList []PortBinding) ContainerBuilder
+	WithPortRange(portRanges []PortRange) ContainerBuilder
 	WithMountPoints(mounts []mount.Mount) ContainerBuilder
 	WithName(name string) ContainerBuilder
 	WithNetworkMode(networkMode string) ContainerBuilder
 	WithEnvironment(envList string) ContainerBuilder
 	WithLabels(labels map[string]string) ContainerBuilder
-	WithLogConfig(config v1.LogConfig)
-	WithRegistryAuth(auth util.RegistryAuth) ContainerBuilder
+	WithLogConfig(config container.LogConfig)
+	WithRegistryAuth(auth RegistryAuth) ContainerBuilder
 	WithAutoRemove(remove bool) ContainerBuilder
-	WithRestartPolicy(policy v1.RestartPolicyName) ContainerBuilder
+	WithRestartPolicy(policy RestartPolicyName) ContainerBuilder
 	WithEntrypoint(cmd []string) ContainerBuilder
 	WithCmd(cmd []string) ContainerBuilder
 	WithUser(uid string) ContainerBuilder
+	WithLogger(logger io.StringWriter) ContainerBuilder
```

Comment:
WithLogWriter would be a tiny bit better name for it, it is a writer after all.

Reply:
Maybe `WithBuilderLogWriter`? This would show that the logger is only used by the builder not by the container.